src/parallel_simulation/splittable_population.rs

```diff
@@ -5,17 +5,18 @@ use crate::parallel_simulation::id_mapping::{MatsimIdMapping, MatsimIdMappings};
 
 use crate::io::matsim_id::MatsimId;
 use crate::parallel_simulation::network::partitioned_network::Network;
-use std::collections::HashMap;
+use std::collections::BTreeMap;
 
 #[derive(Debug)]
 pub struct Population {
-    pub agents: HashMap<usize, Agent>,
+    // To ensure deterministic results when iterating over entries for inserting agents in the queue
```

Comment:
Do you need deterministic iteration, so that the event files are comparable?

Reply:
Exactly. Otherwise, if agents wakeup at the exact same time and are pushed on the queue of the same link, their order is non deterministic and the output changes.